lib/AST/DeclContext.cpp

```diff
@@ -1025,6 +1025,9 @@ DeclContextKind DeclContext::getContextKind() const {
 }
 
 bool DeclContext::hasValueSemantics() const {
+	if (!isTypeContext())
+		return false;
```

Comment:
Looks like some tabs snuck in.

Reply:
Whoops! I’ll fix it after the test finishes.

Reply:
Still not fixed!

Reply:
Though I'm not sure I like the idea of testing it here, rather than asserting it and having callers check.

Reply:
`getSelfTypeInContext` already asserts that, but seems like this check is called from places where we don’t have a type context and thus end up firing the assert (ex: when building the stdlib). 

The code before was calling `getDeclaredInterfaceType` which would just return an `ErrorType` or null if we couldn’t get the type.

Reply:
> `getSelfTypeInContext` already asserts that, but seems like this check is called from places where we don’t have a type context and thus end up firing the assert (ex: when building the stdlib).

Yeah, I remember seeing you add it in here. But asking `hasValueSemantics` doesn't actually make sense for a non-type context, and I don't think we should silently accept it by saying "no".

Reply:
Alright, I have removed the check.